tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Tensors.java

```diff
@@ -0,0 +1,186 @@
+package org.tensorflow;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.StringJoiner;
+import org.tensorflow.ndarray.NdArray;
+import org.tensorflow.ndarray.Shape;
+import org.tensorflow.proto.framework.DataType;
+
+/**
+ * Tensor helper methods.
+ */
+final class Tensors {
+
+  /**
+   * Prevent construction.
+   */
+  private Tensors() {
+  }
+
+  /**
+   * Equivalent to {@link #toString(Tensor, Integer) toString(tensor, null)}.
+   *
+   * @param tensor a tensor
+   * @return the String representation of the tensor
+   */
+  public static String toString(Tensor tensor) {
+    return toString(tensor, null);
+  }
+
+  /**
+   * Returns a String representation of a tensor's data. If the output is wider than {@code
+   * maxWidth} characters, it is truncated and "{@code ...}" is inserted in place of the removed
+   * data.
+   *
+   * @param tensor   a tensor
+   * @param maxWidth the maximum width of the output in characters ({@code null} if unlimited). This
+   *                 limit may surpassed if the first or last element are too long.
+   * @return the String representation of the tensor
+   */
+  public static String toString(Tensor tensor, Integer maxWidth) {
```

Comment:
I'm not sure about having that many ways to print a tensors (if we can call `tensor.toString`, why having `Tensors.toString(tensor)`?) It's a bit confusing and I find that that feature maybe spreads out a bit in multiple parts of the code.

I have a new design to propose, let me know what you think. But what about having this `toString` logic in another called, let say, `TensorPrinter`, which can be returned or directly invoked in the `Tensor` class? i.e. 

```java
class TensorPrinter {

    TensorPrinter(Tensor tensor, int maxWidth) { ... }

    TensorPrinter withMaxWidth(int maxWidth) {
        return new TensorPrinter(this.tensor, maxWidth);
    }

    String print() {
        return .... (this logic here)
    }
}

interface Tensor {

    String print() {
        return new TensorPrinter(this, null).print();
    }

    TensorPrinter printer() {
        return new TensorPrinter(this, null);
    }
}

Tensor t = TFloat32.scalarOf(10.0f);
t.print();
t.printer().maxWidth(10).anotherOption(234).print();
```

Reply:
I prefer your proposal. It is exactly what I was hoping to implement in the long term but I ended up pushing what you see above as a stepping stone in the right direction.

Also, consider the relationship (if any) between `Ops.print()` and this functionality. I may be wrong, but I believe the C++ definition of `tf.print()` maps to what I'm trying to implement in this PR. Does it make sense to have `Ops.print()` delegate to this as well?

Reply:
Do we want to call it `print()` and not `toString()`?

Reply:
If we do that, then I would then expect it to work the same as `Ops.print()` since the two share the same name. Are we planning to merge the two?

Reply:
I think `Ops.print()` simply writes a given string to an output stream, like the console, when the graph is being executed. 

I picked the "print" expression in this example just to show the logic but it can for sure be named otherwise to avoid the confusion. Maybe `TensorStringifier`, `tensor.stringify()` and `tensor.stringifier()`? Or the methods could remain `dataToString()` as well.